internal/vm-console-proxy-bundle/bundle.go

```diff
@@ -27,117 +28,103 @@ var (
 )
 
 type Bundle struct {
-	ServiceAccount     core.ServiceAccount
-	ClusterRole        rbac.ClusterRole
-	ClusterRoleBinding rbac.ClusterRoleBinding
-	Service            core.Service
-	Deployment         apps.Deployment
-	ConfigMap          core.ConfigMap
+	ServiceAccount     *core.ServiceAccount
+	ClusterRole        *rbac.ClusterRole
+	ClusterRoleBinding *rbac.ClusterRoleBinding
+	Service            *core.Service
+	Deployment         *apps.Deployment
+	ConfigMap          *core.ConfigMap
 }
 
-func ReadBundle(path string) (Bundle, error) {
-	files, err := readFile(path)
+func ReadBundle(path string) (*Bundle, error) {
+	data, err := os.ReadFile(path)
 	if err != nil {
-		return Bundle{}, err
+		return nil, err
 	}
 
-	bundleObj, err := decodeObjectsFromFiles(files)
+	bundle, err := loadBundleFromBytes(data)
 	if err != nil {
-		return Bundle{}, err
+		return nil, err
 	}
 
-	return *bundleObj, nil
+	if err = validateBundle(bundle); err != nil {
+		return nil, err
+	}
+	return bundle, nil
 }
 
 func GetBundlePath() string {
 	return filepath.Join(vmConsoleProxyBundleDir, "vm-console-proxy.yaml")
 }
 
-func readFile(fileName string) ([][]byte, error) {
-	file, err := os.ReadFile(fileName)
-	if err != nil {
-		return nil, err
-	}
+func loadBundleFromBytes(data []byte) (*Bundle, error) {
+	bundle := &Bundle{}
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 1024)
+	for {
+		obj := &unstructured.Unstructured{}
+		err := decoder.Decode(&obj)
+		if err == io.EOF {
+			return bundle, nil
+		}
+		if err != nil {
+			return nil, err
+		}
 
-	return [][]byte{file}, nil
-}
+		kind := obj.GetKind()
+
+		var destObj any
+		switch obj.GetKind() {
```

Comment:
`switch kind := obj.GetKind(); kind`?

Reply:
Or if you need the `kind` var later, use it here too?

Reply:
https://github.com/kubevirt/tekton-tasks-operator/blob/main/pkg/tekton-bundle/tekton-bundle.go#L150

Reply:
@0xFelix Done

@ksimon1 That code was used originally here, and I just refactored it to make it smaller.